Automattic/woocommerce-payments
# Move class name to the outer level

#### Changes proposed in this Pull Request

On page load, if any other payment method is selected apart from debit/credit card, the checkbox is hidden. But on selecting the card method, multiple checkboxes are rendered.

![before](https://user-images.githubusercontent.com/33387139/152773381-4b23cf1b-3609-43e9-abcc-a2e6e598b15a.gif)

The reason is the class name is attached to the react [component](https://github.com/Automattic/woocommerce-payments/blob/develop/client/components/platform-checkout/save-user/checkout-page-save-user.js#L30). When another payment method is selected, on window load [this condition](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/platform-checkout/index.js#L17) is false, so the component is rendered but [returns null](https://github.com/Automattic/woocommerce-payments/blob/develop/client/components/platform-checkout/save-user/checkout-page-save-user.js#L26) as the payment method is something else. On ajax request completion [this condition](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/platform-checkout/index.js#L17) is false again as the previous component render did not have the class name (but the component is actually attached already). So it attaches the component again. Finally on selecting card method it shows 2/3 checkboxes.

#### Testing instructions

Recreate the issue-
- First place an order with `cod` method (this will set the `cod` as payment method by default in the next order and help reproduce the issue)
- Add some products from the shop and go to the checkout page. The payment method is set to `cod` so the checkbox is absent
- Select card payment. Multiple checkboxes will appear.

This branch-
- Checkout this branch
- - Add some products from the shop and go to the checkout page. The payment method is set to `cod` so the checkbox is absent
- Select card payment. Only one checkbox will appear as expected

![after](https://user-images.githubusercontent.com/33387139/152775230-e2bde829-b2ec-4e0b-aa78-1f044873d5d6.gif)
